platform/mbed_error.h

```diff
@@ -60,7 +67,17 @@
 extern "C" {
 #endif
 
-void error(const char* format, ...);
+
+static inline void error(const char *format, ...) {
+#ifndef NDEBUG
+    va_list arg;
+    va_start(arg, format);
+    mbed_error_vfprintf(format, arg);
+    va_end(arg);
+#endif
+    while (1) {}
```

Comment:
This instead of ``exit()`` - reason? I cant find it in the commit message. 

We do not care what ``exit()`` does? just a trap here?